### PR TITLE
feat(charges): payment-aware due-date + backfill payment_months (THI-329)

### PR DESCRIPTION
**Pull request by thierryvm**
## PR-A — Payment-aware due-date + backfill payment_months (THI-329)

Corrige à la racine les dates de factures incohérentes, **sans toucher** `nextDueDateForCharge`. Fondation de l'epic « Factures cohérentes » (spec : `docs/plans/epic-factures-coherentes-spec.md`).

### Deux leviers
1. **Migration `20260605000001`** — recompute `payment_months` depuis `(frequency, due_month)` pour les charges incohérentes. Corrige le bug de `20260503000002` (l.58) qui stockait UN seul mois pour les périodiques (S.W.D.E `quarterly` = `[1]` → « janv. 2027 »). **Idempotente, zéro-perte, aucune RLS touchée** (SQL = `paymentMonthsFromFrequency` au caractère près).
2. **Domaine `nextUnpaidDueDate()`** (pur, nouveau) — prochaine occurrence **NON PAYÉE** : saute les payées (Impôt juin payé → roule à juillet) + surface une occurrence du mois courant passée-non-payée en **overdue** (Taxe voiture juin → en retard, plus caché un an). Câblé dans le dashboard `getUpcomingCharges`. La page charges (`ChargesClient`) suit en **PR-B**.

### ⚠️ Migration sur données prod
Le merge applique le backfill sur les `charges` de tous les workspaces. C'est **correctif + idempotent + zéro-perte** (ne touche que les lignes incohérentes, aucun DELETE). Il **dé-casse** la cadence (un seul mois → cadence complète) mais ne corrige PAS l'ancre `due_month` mal saisie (ex. S.W.D.E réel = mai) → @thierry corrige l'ancre via CadenceField (PR-D).

### Gouvernance
plan-reviewer 🟡→intégré · **rls-flow-tester GO** · **financial-formula-validator GO** · 32 tests domaine + suite complète **1433** + build verts.

### Smoke @thierry (preview)
Dashboard `/app` « Prochaines factures » : Impôt (juin payé) roulé à juillet ; Taxe voiture (juin non payée) en « En retard » ; S.W.D.E n'affiche plus janv. 2027.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Introduire un résolveur de date d’échéance sensible aux paiements pour les prochaines facturations et rétro-remplir les cadences de facturation incohérentes afin de rétablir des calendriers de facturation corrects.

Nouvelles fonctionnalités :
- Ajouter une fonction de domaine pour calculer la prochaine date d’échéance impayée d’une facturation, en tenant compte de l’historique des paiements et en mettant en évidence les occurrences en retard.
- Connecter le pipeline du tableau de bord `upcoming-charges` au nouveau résolveur de date d’échéance sensible aux paiements, tout en préservant l’API publique existante.

Corrections de bugs :
- Rétro-remplir les `payment_months` incorrects pour les facturations périodiques afin de restaurer les cadences complètes dérivées de `frequency` et `due_month`, corrigeant les cas où les prochaines dates d’échéance étaient résolues avec un an d’avance.
- S’assurer que les compartiments de retard incluent désormais correctement les occurrences du mois en cours qui sont échues mais non payées, au lieu de les faire avancer silencieusement.

Améliorations :
- Affiner le comportement de `upcoming-charges` pour ignorer les occurrences déjà payées de la période en cours et supprimer les entrées qui dépassent l’horizon de 30 jours, sans modifier les interfaces existantes.
- Documenter le nouveau comportement de date d’échéance sensible aux paiements et le plan de migration dans un document de conception dédié.

Documentation :
- Ajouter un plan de conception et d’implémentation pour la date d’échéance sensible aux paiements et le travail de rétro-remplissage de cadence, y compris l’évaluation des risques et les invariants.

Tests :
- Ajouter des tests unitaires pour le nouveau résolveur de prochaine date d’échéance impayée sensible aux paiements et mettre à jour les tests de `upcoming-charges` pour couvrir la gestion des retards, le saut des paiements et les cas limites de calendrier.

Tâches diverses :
- Introduire une migration Supabase corrective et idempotente pour recalculer `payment_months` en utilisant des règles de cadence canoniques pour toutes les facturations existantes.

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Introduce a payment-aware due-date resolver for upcoming charges and backfill inconsistent charge cadences to restore correct billing schedules.

New Features:
- Add a domain function to compute the next unpaid due date for a charge, taking into account payment history and surfacing overdue occurrences.
- Wire the dashboard upcoming-charges pipeline to use the new payment-aware due-date resolver while preserving the existing public API.

Bug Fixes:
- Backfill incorrect payment_months for periodic charges to restore full cadences derived from frequency and due_month, fixing cases where next due dates were resolved a year ahead.
- Ensure overdue buckets now correctly include passed-but-unpaid current-month occurrences instead of silently rolling them forward.

Enhancements:
- Refine upcoming-charges behavior to skip already-paid current-period occurrences and drop entries that fall beyond the 30-day horizon without changing existing interfaces.
- Document the new payment-aware due-date behavior and migration plan in a dedicated design plan document.

Documentation:
- Add a design and implementation plan for the payment-aware due-date and cadence backfill work, including risk assessment and invariants.

Tests:
- Add unit tests for the new payment-aware next-unpaid-due-date resolver and update upcoming-charges tests to cover overdue handling, payment skipping, and calendar edge cases.

Chores:
- Introduce a corrective, idempotent Supabase migration to recompute payment_months using canonical cadence rules for all existing charges.

</details>

### CODE REVIEW

File: src/lib/domain/charges/next-unpaid-due-date.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  const refYear = Number(refYearStr);
+  const refMonth = Number(refMonthStr);
+  if (!Number.isFinite(refYear) || !Number.isFinite(refMonth)) return null;
+
+  const monthsSet = new Set(charge.paymentMonths);
+
+  // Search the current month and the next 23 — covers every cadence (monthly
+  // resolves at offset 0, annual within 12); the 24-cap is defensive against
+  // an ill-formed `paymentMonths`.
+  for (let offset = 0; offset < 24; offset += 1) {
```
[the remark this answers is not in the document]
Comment (thierryvm) [reply src/lib/domain/charges/next-unpaid-due-date.ts:65]: Acknowledged — deferring this refactor (advisory complexity, not a bug). The 3 date helpers (daysInMonth/pad2/pad4) intentionally mirror next-due-date.ts for pure-domain module isolation (each resolver is self-contained). financial-formula-validator reviewed this and endorsed it as acceptable, recommending a shared date-util extraction only once a 3rd resolver appears. Merging the two offset loops would over-abstract two genuinely different algorithms (skip-past-day vs skip-paid). next-due-date.ts is explicitly out of PR-A scope (banned-list). Tracked as a follow-up cleanup.
